test/server.test.js

```diff
@@ -1,14 +1,262 @@
+const products = require('../initial-data/products.json');
 
-describe('Brands', () => {});
+// TESTS FOR THE SERVER
+describe('Home', () => {
+  describe('GET /', () => {
+    it('it should return status 200', (done) => {
+      chai.request(server)
+        .get('/')
+        .end((err, res) => {
+          if (err) return done(err);
+          expect(res).to.have.status(200);
+          done();
+        });
+    });
+    it('it should return status 500 if an error occurs', (done) => {
+      const originalRequest = chai.request(server).get;
 
-describe('Login', () => {});
+      chai.request(server).get = () => {
+        return Promise.reject(new Error('Server error blah blah'));
+      };
 
-describe('Cart', () => {});
+      chai.request(server)
+        .get('/')
+        .end((err, res) => {
+          chai.request(server).get = originalRequest;
+
+          if (err) {
+            expect(res).to.have.status(500);
+          } else {
+            expect(res).to.have.status(200);
+          }
+          done();
+        });
+    });
+  });
+});
+
+describe('Brands', () => {
+  describe('GETS /brands', () => {
+    it('it should return all FIVE brand names, including example brand name [0] "Burberry"', (done) => {
+      chai.request(server)
+        .get('/brands')
+        .end((err, res) => {
+          if (err) return done(err);
+          expect(res).to.have.status(200);
+          expect(res.body).to.be.an('object');
+          expect(res.body['All Brand Names']).to.have.lengthOf(5);
+          expect(res.body['All Brand Names'][0]).to.equal('Burberry');
+          done();
+        });
+    });
+  });
+  describe('GETS /brands/:name', () => {
+    it('it should return product names by brand', (done) => {
+      const productName = 'Burberry';
+      chai.request(server)
+        .get(`/brands/${productName}`)
+        .end((err, res) =>{
+          if (err) return done(err);
+          expect(res).to.have.status(200);
+          expect(res.body).to.be.an('object');
+          expect(res.body).to.have.property('Burberry');
+          
+          const products = res.body['Burberry'];
+
+          expect(products).to.be.an('array');
+
+          products.forEach(product => {
+            expect(product).to.have.property('id');
+            expect(product).to.have.property('categoryId');
+            expect(product).to.have.property('name');
+            expect(product).to.have.property('description');
+            expect(product).to.have.property('price');
+            expect(product).to.have.property('imageUrls');
+          })
+          done();
+        });
+    });
+  });
+});
+
+describe('Products', () => {
+  describe('GETS /products', () => {
+    it('it should return all ELEVEN product names', (done) => {
+      chai.request(server)
+        .get('/products')
+        .end((err, res) => {
+          if (err) return done(err);
+          expect(res).to.have.status(200);
+          expect(res.body).to.be.an('object');
```

Comment:
this is not really valuable for testing. 
Becomes redundant for the next steps you do in the test